distributor/src/distributor/messenger.bal

```diff
@@ -14,26 +12,53 @@ twilio:TwilioConfiguration twilioConfig = {
 twilio:Client twilioClient = new(twilioConfig);
 
 // Keeps registered sms recipients in-memory. Values are populated in every service init and recipient registration
-string[] mobileSubscribers = [];
+map<string> mobileSubscribers = {};
 string sourceMobile = config:getAsString("eclk.sms.twilio.source");
 boolean validTwilioAccount = false;
 
 # Send SMS notification to all the subscribers.
 #
-# + electionCode - The respective code that represents the type of election
-# + resultCode - The predefined code for a released result
-function sendSMS(string electionCode, string resultCode) {
-    string division = DIVISION_CODES[resultCode] ?: resultCode;
-    string message  = "Await results for " + electionCode +  "/" + division + "(" + resultCode + ")";
-
-    foreach string targetMobile in mobileSubscribers {
-        if (targetMobile == INVALID_NO) {
-            continue;
+# + notification - The result notification record
+function sendSMS(Notification notification) {
+    string electionCode = notification.electionCode;
+    string electionType = "/" + notification.'type;
+    string resultCode = notification.resultCode;
+    string level = notification.level;
+
+    string message = "";
+
+    match level {
+        LEVEL_PD => {
+            string electoralDistrict = "/" + notification["ed_name"].toString();
+            string pollingDivision = "/" + notification["pd_name"].toString();
+
+            message  = "Await POLLING-DIVISION results for " + electionCode + electionType + electoralDistrict +
+                        pollingDivision;
+        }
+        LEVEL_ED => {
+            string electoralDistrict = "/" + notification["ed_name"].toString();
+
+            message  = "Await ELECTORAL-DISTRICT results for " + electionCode + electionType + electoralDistrict;
+        }
+        LEVEL_NF => {
+            message  = "Await NATIONAL-FINAL results for " + electionCode + electionType;
         }
+        _ => {
+            message  = "Await results for " + electionCode + electionType + "/" + resultCode;
+        }
+    }
+
+    message = message + "(" + resultCode + ")";
+
+    map<string> currentMobileSubscribers = mobileSubscribers;
```

Comment:
Do we need to assign this to a local variable? The reference is still of the original value?